RackApp/app.rb

```diff
@@ -0,0 +1,71 @@
+require "erb"
+require "rack"
+require "active_support/time"
+
+class MyController
+  attr_reader :person, :time
+
+  def initialize(user, time)
+    @person = user
+    @time = time
+  end
+
+  def file
+    File.read(File.join(__dir__, "template.html.erb"))
+  end
+
+  def template
+    ERB.new(file)
+  end
+
+  def render
+    template.result(binding)
+  end
+end
+
+class User
+  attr_reader :nickname
+
+  def initialize(nickname)
+    @nickname = nickname
+  end
+end
+
+class CountryTime
+  attr_reader :zone, :timezone
+
+  def initialize(zone)
+    @zone = zone
+  end
+
+  def timezone
+    Time.zone = @zone
+    @time = Time.zone.now
+  end
+end
+
+class Display
+  def initialize(app)
+    @app = app
+  end
+
+  def self.call(env)
+    req = Rack::Request.new(env)
+    nickname = req.params["nickname"]
+    tz = req.params["tz"]
+    if req.path.match("/users") && nickname && tz
+      user = User.new("#{nickname}")
```

Comment:
You can avoid String interpolation by using above `nickane = params['nickname'].to_s` and son on